packages/starlight/user-components/Badge.astro

```diff
@@ -0,0 +1,143 @@
+---
+import { AstroError } from 'astro/errors';
+import { type Badge, badgeVariants } from '../schemas/badge';
+
+interface Props {
+	variant?: Badge['variant'];
+	class: string | undefined;
+}
+
+const { variant = 'default', class: className } = Astro.props;
+
+if (!badgeVariants.includes(variant)) {
+	throw new AstroError(
+		'Invalid `variant` prop passed to the `<Badge>` component.\n',
+		`Received: ${JSON.stringify(variant)}\n` +
+			`Expected one of ${badgeVariants.map((i) => JSON.stringify(i)).join(', ')}`
+	);
+}
+---
+
+<span class:list={['sl-badge', variant, className]}><slot /></span>
```

Comment:
I agree, after taking a look after some time I think it's better to have a text prop. My previous idea was to pass in a slot in case users would like to pass svgs or other elements but I would like to keep it simple and keep it just with text.